#! /bin/sh /usr/share/dpatch/dpatch-run
## 50-gcc43-headers.dpatch by Brian M. Carlson <sandals@crustytoothpaste.ath.cx>
##
## DP: Fix build failure with GCC 4.3.

@DPATCH@
diff -urNad libmusicbrainz-2.1-2.1.5~/lib/c_wrapper.cpp libmusicbrainz-2.1-2.1.5/lib/c_wrapper.cpp
--- libmusicbrainz-2.1-2.1.5~/lib/c_wrapper.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1-2.1.5/lib/c_wrapper.cpp	2008-03-03 22:02:29.000000000 +0100
@@ -25,6 +25,8 @@
 #include "trm.h"
 #include "mb_c.h"
 
+#include <string.h>
+
 extern "C"
 {
 
diff -urNad libmusicbrainz-2.1-2.1.5~/lib/comhttpsocket.cpp libmusicbrainz-2.1-2.1.5/lib/comhttpsocket.cpp
--- libmusicbrainz-2.1-2.1.5~/lib/comhttpsocket.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1-2.1.5/lib/comhttpsocket.cpp	2008-03-03 22:01:20.000000000 +0100
@@ -19,6 +19,7 @@
 #include "comsocket.h"
 #endif
 #include <stdlib.h>
+#include <string.h>
 #include <assert.h>
 
 const char* g_strCOMVer = "libmusicbrainz/"VERSION;
diff -urNad libmusicbrainz-2.1-2.1.5~/lib/comsocket.cpp libmusicbrainz-2.1-2.1.5/lib/comsocket.cpp
--- libmusicbrainz-2.1-2.1.5~/lib/comsocket.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1-2.1.5/lib/comsocket.cpp	2008-03-03 22:01:58.000000000 +0100
@@ -37,6 +37,7 @@
 #include <netinet/tcp.h>
 #include <errno.h>
 #include <stdio.h>
+#include <string.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
diff -urNad libmusicbrainz-2.1-2.1.5~/lib/http.cpp libmusicbrainz-2.1-2.1.5/lib/http.cpp
--- libmusicbrainz-2.1-2.1.5~/lib/http.cpp	2007-05-11 08:21:21.000000000 +0200
+++ libmusicbrainz-2.1-2.1.5/lib/http.cpp	2008-03-03 22:02:55.000000000 +0100
@@ -24,6 +24,7 @@
 ----------------------------------------------------------------------------*/
 #include <stdlib.h>
 #include <stdio.h> 
+#include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <ctype.h>
diff -urNad libmusicbrainz-2.1-2.1.5~/lib/sigclient.cpp libmusicbrainz-2.1-2.1.5/lib/sigclient.cpp
--- libmusicbrainz-2.1-2.1.5~/lib/sigclient.cpp	2006-11-14 17:34:49.000000000 +0100
+++ libmusicbrainz-2.1-2.1.5/lib/sigclient.cpp	2008-03-03 22:03:50.000000000 +0100
@@ -38,6 +38,8 @@
 
 #include "sigxdr.h"
 
+#include <cstring>
+
 char tooShortTRM[] = { 0xf9, 0x80, 0x9a, 0xb1, 0x2b, 0x0f, 0x4d, 0x78, 
                        0x88, 0x62, 0xfb, 0x42, 0x5a, 0xde, 0x8a, 0xb9 };
 char sigserverBusyTRM[] = { 0xc4, 0x57, 0xa4, 0xa8, 0xb3, 0x42, 0x4e, 0xc9, 
